src/ReplicatedStorage/Client/InstanceComponents/Misc/Jetpack.lua

```diff
@@ -0,0 +1,152 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+local JetpackState = require(ReplicatedStorage.Common.Util.JetpackState)
+
+--[=[
+	@class Jetpack
+]=]
+local Jetpack = {}
+Jetpack.__index = Jetpack
+
+function Jetpack.new(instance)
+	local clientComm = Comm.ClientComm.new(instance, false, "Jetpack")
+	local self = setmetatable({}, Jetpack)
+
+	self.Instance = instance
+	self._comm = clientComm
+	self._trove = Trove.new()
+
+	-- Set owning player
+	self._owner = Players:GetPlayerFromCharacter(instance.Parent)
+	instance.AncestryChanged:Connect(function()
+		self._owner = Players:GetPlayerFromCharacter(instance.Parent)
+	end)
+	if not self:IsOwner() then
+		return self
+	end
+
+	-- Create state
+	self._state = JetpackState.new({
+		Boosting = instance:GetAttribute("Boosting");
+
+		Fuel = instance:GetAttribute("Fuel");
+		Capacity = instance:GetAttribute("Capacity");
+		FillRate = instance:GetAttribute("FillRate");
+		BurnRate = instance:GetAttribute("BurnRate");
+	})
+
+	-- Add state and comm to trove
+	self._trove:Add(self._state)
+	self._trove:Add(self._comm)
+
+	-- Create server object
+	self._serverObject = clientComm:BuildObject()
+
+	-- Subscribe to attribute changes
+	self._trove:Connect(instance:GetAttributeChangedSignal("Boosting"), function()
+		self._state._silo:Dispatch(self._state._silo.Actions.SetBoosting(instance:GetAttribute("Boosting")))
+	end)
+	self._trove:Connect(instance:GetAttributeChangedSignal("Fuel"), function()
+		self._state._silo:Dispatch(self._state._silo.Actions.SetFuel(instance:GetAttribute("Fuel")))
+	end)
+	self._trove:Connect(instance:GetAttributeChangedSignal("Capacity"), function()
+		self._state._silo:Dispatch(self._state._silo.Actions.SetCapacity(instance:GetAttribute("Capacity")))
+	end)
+	self._trove:Connect(instance:GetAttributeChangedSignal("FillRate"), function()
+		self._state._silo:Dispatch(self._state._silo.Actions.SetFillRate(instance:GetAttribute("FillRate")))
+	end)
+	self._trove:Connect(instance:GetAttributeChangedSignal("BurnRate"), function()
+		self._state._silo:Dispatch(self._state._silo.Actions.SetBurnRate(instance:GetAttribute("BurnRate")))
```

Comment:
These should probably all be methods of JetpackState - you're modifying the internal workings of a JetpackState in it's dependent file - which is very inappropriate OOP